internal/handlers/nginx_config_post_handler.go

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright Steve Wagner. All rights reserved.
+ * Use of this source code is governed by the Apache License that can be found in the LICENSE file.
+ */
+
+package handlers
+
+import (
+	"ciroque/go-http-server/internal/metrics"
+	"ciroque/go-http-server/internal/responses"
+	"encoding/json"
+	"fmt"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"io"
+	"net/http"
+)
+
+type NginxConfigPostHandler struct {
+	Context          *Context
+	RequestCount     *prometheus.CounterVec
+	RequestDurations *prometheus.HistogramVec
+}
+
+func NewNginxConfigPostHandler(context *Context) *NginxConfigPostHandler {
+	return &NginxConfigPostHandler{Context: context}
+}
+
+func (handler *NginxConfigPostHandler) Cleanup() {
+	handler.Context.Logger.Info("NginxConfigPostHandler::Cleanup")
+	prometheus.Unregister(handler.RequestCount)
+	prometheus.Unregister(handler.RequestDurations)
+}
+
+func (handler *NginxConfigPostHandler) BuildHandler() (http.Handler, error) {
+	handler.configureMetrics()
+	err := handler.registerMetrics()
+	if err != nil {
+		return nil, err
+	}
+
+	h := promhttp.InstrumentHandlerCounter(
+		handler.RequestCount,
+		promhttp.InstrumentHandlerDuration(
+			handler.RequestDurations,
+			http.HandlerFunc(handler.handler)))
+
+	return h, nil
+}
+
+func (handler *NginxConfigPostHandler) Route() string {
+	return "/nginx/config"
+}
+
+func (handler *NginxConfigPostHandler) handler(writer http.ResponseWriter, request *http.Request) {
+	responseStatusCode := http.StatusAccepted
+	if request.Method != http.MethodPost {
+		writer.WriteHeader(http.StatusMethodNotAllowed)
+		return
+	}
+
+	body, err := io.ReadAll(request.Body)
+	if err != nil {
+		handler.writeErrorResponse(writer, "Error reading request body", http.StatusInternalServerError)
+		return
+	}
+
+	if len(body) == 0 {
+		handler.writeErrorResponse(writer, "An NGINX configuration was not found in the request", http.StatusBadRequest)
+		return
+	}
+
+	eventId, configUpdateEvent, err := handler.Context.Processor.QueueConfigUpdate(body)
+	if err != nil {
+		handler.Context.Logger.Warnf("Error queuing the NGINX configuration, %#v", err)
+		responseStatusCode = http.StatusConflict
+	}
+
+	location := fmt.Sprintf("/nginx/config/status/%d", eventId)
+	writer.Header().Add("Location", location)
+
+	// Close the request body to release resources
+	defer request.Body.Close()
+
+	writer.WriteHeader(responseStatusCode)
+	eventJson, _ := json.Marshal(configUpdateEvent)
```

Comment:
This should be a method on the update config module; to return the json representation...